firedrake/cython/dmcommon.pyx

```diff
@@ -2552,7 +2553,7 @@ cdef struct CommFacet:
     PetscInt global_u, global_v
     PetscInt local_facet
 
-cdef int CommFacet_cmp(const void *x_, const void *y_) nogil:
+cdef int CommFacet_cmp(void *x_, void *y_) noexcept nogil:
```

Comment:
I think you did not intend to remove `const`s here?

Reply:
@connorjward is that right? I think you made this commit

Reply:
I don't really remember exactly why I did this. I tackled a bunch of signature changes because Cython 3 is a little stricter. I think it's likely that this was a necessary change but I'm not 100% sure.

Reply:
This is something explicitly fixed in https://github.com/firedrakeproject/firedrake/pull/3503/files, so, maybe, when Connor started making Cython3 changes, this fix was not in, and later merge/rebase process somehow dropped `const`s.

Reply:
Ah I see. We both modified the same line. Good catch.